src/commands/git.rs

```diff
@@ -624,12 +629,31 @@ fn cmd_git_push(
         get_default_push_remote(ui, command, &git_repo)?
     };
 
-    let mut tx;
+    let repo = workspace_command.repo().clone();
+    let wc_commit_id = workspace_command.get_wc_commit_id().cloned();
+    let change_commits: Vec<_> = args
+        .change
+        .iter()
+        .map(|change_str| workspace_command.resolve_single_rev(change_str))
+        .try_collect()?;
+    let revision_commits = resolve_multiple_nonempty_revsets(&args.revisions, &workspace_command)?;
+    fn find_branches_targeting<'a>(
+        view: &'a View,
+        target: &RefTarget,
+    ) -> Vec<(&'a String, &'a BranchTarget)> {
+        view.branches()
+            .iter()
+            .filter(|(_, branch_target)| branch_target.local_target.as_ref() == Some(target))
+            .collect()
+    }
+
+    let mut tx = workspace_command.start_transaction("");
```

Comment:
Maybe an helper would be nice. E.g `workspace_command.start_empty_transaction()`?

Reply:
Both options seem good, but let's wait for more users.